cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/config/ETLBatchConfig.java

```diff
@@ -29,34 +31,49 @@
 public final class ETLBatchConfig extends ETLConfig {
   private final String schedule;
   private final List<ETLStage> actions;
+  private final List<LocalizeResourceInfo> resourcesToLocalize;
```

Comment:
Thought about that, but did it this way so that the same local file can be shared by multiple plugins. Theoretically, even if a plugin localizes a file, it can be made available to the entire pipeline. However, seemed like for that use case, explicitly making it a pipeline level configuration made sense. Otherwise, if pluginA localizes a file with the name "file1.txt", then we'll have to set the file name that pluginA has localized in pluginB's config. Because of this, it looked cleaner to me if this was a pipeline level configuration.

Come to think of it, maybe we should also support the use case that a single plugin wants to localize a file and use it. Other plugins do not care about it. That would mean there are two sets of localized files: 
1. Globally localized using Pipeline configuration, available to all plugins in the pipeline.
2. Localized by a given plugin. Only accessed by that plugin.